spec/api-session-spec.js

```diff
@@ -635,30 +635,71 @@ describe('session module', () => {
       }
     })
 
-    it('allows configuring proxy settings', (done) => {
+    it('allows configuring proxy settings', async () => {
+      const config = { proxyRules: 'http=myproxy:80' }
+      await customSession.setProxy(config)
+      const proxy = await customSession.resolveProxy('http://example.com/')
+      assert.strictEqual(proxy, 'PROXY myproxy:80')
+    })
+
+    // TODO(codebytere): remove when Promisification is complete
+    it('allows configuring proxy settings (callback)', (done) => {
       const config = { proxyRules: 'http=myproxy:80' }
       customSession.setProxy(config, () => {
-        customSession.resolveProxy('http://example.com/', (proxy) => {
+        customSession.resolveProxy('http://example.com/', proxy => {
           assert.strictEqual(proxy, 'PROXY myproxy:80')
           done()
         })
       })
     })
 
-    it('allows removing the implicit bypass rules for localhost', (done) => {
+    it('allows removing the implicit bypass rules for localhost', async () => {
       const config = {
         proxyRules: 'http=myproxy:80',
         proxyBypassRules: '<-loopback>'
       }
-      customSession.setProxy(config, () => {
-        customSession.resolveProxy('http://localhost', (proxy) => {
+
+      await customSession.setProxy(config)
+      const proxy = await customSession.resolveProxy('http://localhost')
+      assert.strictEqual(proxy, 'PROXY myproxy:80')
+    })
+
+    // TODO(codebytere): remove when Promisification is complete
+    it('allows removing the implicit bypass rules for localhost (callback)', (done) => {
+      const config = {
+        proxyRules: 'http=myproxy:80',
+        proxyBypassRules: '<-loopback>'
+      }
+      customSession.setProxy(config).then(() => {
+        customSession.resolveProxy('http://localhost').then(proxy => {
           assert.strictEqual(proxy, 'PROXY myproxy:80')
           done()
         })
       })
     })
 
-    it('allows configuring proxy settings with pacScript', (done) => {
+    it('allows configuring proxy settings with pacScript', () => {
```

Comment:
this one probably needs to stay unchanged, the top-level function is not running a Promise resolved when the code in `server.listen` resolves